qualtran/bloqs/qubitization/qubitization_walk_operator.py

```diff
@@ -82,66 +86,67 @@ class QubitizationWalkOperator(SpecializedSingleQubitControlledGate):
         Babbush et. al. (2018). Figure 1.
     """
 
-    select: SelectOracle
-    prepare: PrepareOracle
+    block_encoding: BlockEncoding
     control_val: Optional[int] = None
     uncompute: bool = False
 
-    def __attrs_post_init__(self):
-        assert self.select.control_registers == self.reflect.control_registers
-
     @cached_property
     def control_registers(self) -> Tuple[Register, ...]:
-        return self.select.control_registers
+        return self.block_encoding.control_registers
 
     @cached_property
     def selection_registers(self) -> Tuple[Register, ...]:
-        return self.prepare.selection_registers
+        return self.block_encoding.selection_registers
 
     @cached_property
     def target_registers(self) -> Tuple[Register, ...]:
-        return self.select.target_registers
+        return self.block_encoding.target_registers
+
+    @cached_property
+    def junk_registers(self) -> Tuple[Register, ...]:
+        return self.block_encoding.junk_registers
 
     @cached_property
     def signature(self) -> Signature:
         return Signature(
-            [*self.control_registers, *self.selection_registers, *self.target_registers]
+            [
+                *self.control_registers,
+                *self.selection_registers,
+                *self.target_registers,
+                *self.junk_registers,
+            ]
         )
 
     @cached_property
     def reflect(self) -> ReflectionUsingPrepare:
-        return ReflectionUsingPrepare(self.prepare, control_val=self.control_val, global_phase=-1)
+        return ReflectionUsingPrepare(
+            self.block_encoding.signal_state, control_val=self.control_val, global_phase=-1
+        )
 
     @cached_property
     def sum_of_lcu_coefficients(self) -> Optional[SymbolicFloat]:
         r"""value of $\lambda$, i.e. sum of absolute values of coefficients $w_l$."""
-        return self.prepare.l1_norm_of_coeffs
+        return self.block_encoding.alpha
```

Comment:
`QubitizationWalkOperator` is also a block-encoding of `H` right? We should perhaps inherit from `BlockEncoding`, get rid of `sum_of_lcu_coefficients`, and instead always use `walk_op.alpha` (which would be `BH.alpha`).